Source/ACE.Database/Database.cs

```diff
@@ -231,6 +316,82 @@ private void ConstructGetListStatement<T1>(T1 id, Type type)
             PrepareStatement(statementId, query, types);
         }
 
+        private void ConstructDeleteListStatement<T1>(T1 id, Type type)
+        {
+            uint statementId = Convert.ToUInt32(id);
+            DbTableAttribute dbTable = type.GetCustomAttributes(false)?.OfType<DbTableAttribute>()?.FirstOrDefault();
+
+            // XXX(ddevec): Slightly hacky -- I increment the DB field by 200... This elides the need for extra attributes
+            DbGetListAttribute getList = type.GetCustomAttributes(false)?.OfType<DbGetListAttribute>()?.FirstOrDefault(d => d.ConstructedStatementId + 200 == statementId);
+
+            if (dbTable == null)
+                Debug.Assert(false, $"Statement Construction failed for type {type}");
+
+            if (getList == null)
+                Debug.Assert(false, $"Statement Construction failed for type {type}");
+
+            List<MySqlDbType> types = new List<MySqlDbType>();
+            var properties = GetPropertyCache(type);
+            string tableName = getList.TableName;
+            string whereList = null;
+
+            foreach (var p in properties)
+            {
+                if (getList.ParameterFields.Contains(p.Item2.DbFieldName))
+                {
+                    if (whereList != null)
+                        whereList += " AND ";
+                    whereList += "`" + p.Item2.DbFieldName + "` = ?";
+                    types.Add((MySqlDbType)p.Item2.DbFieldType);
+                }
+            }
+
+            string query = $"DELETE FROM `{tableName}` WHERE {whereList}";
+
+            PrepareStatement(statementId, query, types);
+        }
+
+        private void ConstructInsertListStatement<T1>(T1 id, Type type)
+        {
+            uint statementId = Convert.ToUInt32(id);
+            DbTableAttribute dbTable = type.GetCustomAttributes(false)?.OfType<DbTableAttribute>()?.FirstOrDefault();
+
+            // XXX(ddevec): Slightly hacky -- I increment the DB field by 200... This elides the need for extra attributes
```

Comment:
that's not slighly hacky, that's bite-us-in-the-ass-in-under-a-week major hacky :(